File: Problem Set 1/mario.c
#include <stdio.h>

int main(void)
{
    int height;

    // Prompt user for pyramid height until valid input is given
    do
    {
        // Prompt for the height
        printf("Height: ");
        // Check if input is an integer and store it in the variable 'height'
    } while (scanf("%d", &height) != 1 || height < 1 || height > 8);

    // Loop to print each row of the pyramid
    for (int i = 1; i <= height; i++)
    {
        // Print leading spaces
        for (int j = 0; j < height - i; j++)
        {
            printf(" ");
        }

        // Print hashes
        for (int j = 0; j < i; j++)
        {
            printf("#");
        }

        // Newline after each row
        printf("\n");
    }

    return 0;
}

File: Problem Set 6/sentimental-mario-less/mario.py
from cs50 import get_int

# Prompt user until valid input is given
while True:
    height = get_int("Height: ")
    if 1 <= height <= 8:
        break

# Print the pyramid
for i in range(1, height + 1):
    spaces = height - i
    hashes = i
    print(" " * spaces + "#" * hashes)
